open-metrics-report/src/main.rs

```diff
@@ -0,0 +1,201 @@
+                .expect("Should be able to cast broker address to SocketAddr type."),
+        )
+        .install()
+        .expect("prometheus metrics exporter should be setup");
+
+    metrics::describe_counter!(
+        METRIC_MSG_COUNT,
+        metrics::Unit::Count,
+        "Number of messages received"
+    );
+
+    metrics::describe_gauge!(
+        METRIC_LAST_MSG_TIMESTAMP,
+        metrics::Unit::Nanoseconds,
+        "Timestamp of last message received (ns)"
+    );
+
+    metrics::describe_gauge!(
+        METRIC_LAST_MSG_FRAME_NUMBER,
+        metrics::Unit::Count,
+        "Frame number of last message received"
+    );
+
+    metrics::describe_gauge!(
+        METRIC_CHANNEL_COUNT,
+        metrics::Unit::Count,
+        "Number of channels in last message received"
+    );
+
+    metrics::describe_gauge!(
+        METRIC_SAMPLE_COUNT,
+        metrics::Unit::Count,
+        "Number of samples in last message received"
+    );
+
+    task::spawn(poll_kafka_msg(consumer));
+
+    Ok(())
+}
+
+/// Poll kafka messages and update digitiser data.
+async fn poll_kafka_msg(consumer: StreamConsumer) {
+    loop {
+        match consumer.recv().await {
+            Err(e) => log::warn!("Kafka error: {}", e),
+            Ok(msg) => {
+                log::debug!(
+                    "key: '{:?}', topic: {}, partition: {}, offset: {}, timestamp: {:?}",
+                    msg.key(),
+                    msg.topic(),
+                    msg.partition(),
+                    msg.offset(),
+                    msg.timestamp()
+                );
+
+                if let Some(payload) = msg.payload() {
+                    if digitizer_analog_trace_message_buffer_has_identifier(payload) {
+                        match root_as_digitizer_analog_trace_message(payload) {
+                            Ok(data) => {
+                                let id = data.digitizer_id();
+                                let labels = [("dynamic_key", format!("{}", id))];
+
+                                /* Metrics */
+                                // Message recieved count
+                                counter!("digitiser_message_received_count", &labels).increment(1);
+
+                                // Last message frame number
+                                let frame_number = data.metadata().frame_number();
+                                gauge!("digitiser_last_message_frame_number", &labels)
+                                    .set(frame_number as f64);
+
+                                // Channel count
+                                let channel_count = match data.channels() {
+                                    Some(c) => c.len(),
+                                    None => 0,
+                                };
+                                gauge!("digitiser_channel_count", &labels)
+                                    .set(channel_count as f64);
+
+                                // Sample count
```

Comment:
Sample count also has a label of channel number, so all channels should be reported.